tls/extensions/s2n_extension_type_lists.c

```diff
@@ -123,6 +123,10 @@ static const s2n_extension_type *const certificate_extensions[] = {
         &s2n_server_sct_list_extension,
 };
 
+static const s2n_extension_type *const nst_extensions[] = {
```

Comment:
Why don't these extension lists have null terminators?

Reply:
They don't need null terminators. You can get the list size with s2n_array_len(nst_extensions) (basically sizeof(a)/sizeof(a[0])), and the list of lists referenced by code outside this module stores the length of each list as part of its entries.

Reply:
This is how we handle an empty list: https://github.com/aws/s2n-tls/blob/main/tls/extensions/s2n_extension_type_lists.c#L136